mlos_bench/mlos_bench/services/local/local_exec.py

```diff
@@ -120,18 +181,12 @@ def _local_exec_script(self, script_line: str,
         (return_code, stdout, stderr) : (int, str, str)
             A 3-tuple of return code, stdout, and stderr of the script process.
         """
-        cmd = shlex.split(script_line)
-        script_path = self.config_loader_service.resolve_path(cmd[0])
-        # special case handling for leading lone `.` character, which also means `source`
-        # (as in, include, but not execute) in shell syntax
-        if os.path.exists(script_path) and not os.path.isdir(script_path):
-            script_path = os.path.abspath(script_path)
-        else:
-            script_path = cmd[0]  # rollback to the original value
-
-        cmd = [script_path] + cmd[1:]
-        if script_path.strip().lower().endswith(".py"):
-            cmd = [sys.executable] + cmd
+        # Split the command line into set of subcmd tokens.
+        # For each subcmd, perform path resolution fixups for any scripts being executed.
+        subcmds = split_cmdline(script_line)
+        subcmds = [self._resolve_cmdline_script_path(subcmd) for subcmd in subcmds]
+        # Finally recombine all of the fixed up subcmd tokens into the original.
+        cmd = [token for subcmd in subcmds for token in subcmd]
```

Comment:
Could that be the source of the problem? It looks like we try to resolve the paths of _all_ tokens of a command line. Before this change we were doing it only for the script

Reply:
We're still only doing it for the script.

Per this PR, the `split_cmdline` splits complex shell commands like `foo path/to/file && bar || baz --args` into an array of individual commands:
```python
[
  [`foo`, `path/to/file`],
  [`&&`],
  [`bar`],
  [`||`],
  [`baz`, `--args`],
]
```
https://github.com/bpkroth/MLOS/blob/c7fa42b30b05addb3f8236c88e2a33a21b40baf7/mlos_bench/mlos_bench/services/local/local_exec.py#L28

Then `_resolve_cmdline_script_path` only looks at the first item in each subcmd list to resolve it:
https://github.com/bpkroth/MLOS/blob/c7fa42b30b05addb3f8236c88e2a33a21b40baf7/mlos_bench/mlos_bench/services/local/local_exec.py#L136

There's no sane way to do that for arguments after the initial subcommand.
They could be files, string args, etc.